# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1381267097 -7200
#      Tue Oct 08 23:18:17 2013 +0200
# Node ID 07a4597fcbac9b890ae5f517556af5d047c24a8e
# Parent  b43da3876b64f6731a303b4efa3f2b3ab8484dac
graphics.cc: don't ask for (orphan) dummy axes parent properties.

* axes::properties::get_boundingbox : get default figure size if parent
object is invalid (Bug # 39371)

diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -5564,18 +5564,21 @@ axes::properties::get_boundingbox (bool 
                   get_position ().matrix_value ()
                   : get_outerposition ().matrix_value ());
   Matrix parent_size (parent_pix_size);
 
   if (parent_size.numel () == 0)
     {
       graphics_object obj = gh_manager::get_object (get_parent ());
 
-      parent_size =
-       obj.get_properties ().get_boundingbox (true).extract_n (0, 2, 1, 2);
+      if (obj.valid_object ())
+        parent_size =
+          obj.get_properties ().get_boundingbox (true).extract_n (0, 2, 1, 2);
+      else
+        parent_size = default_figure_position ();
     }
 
   pos = convert_position (pos, get_units (), "pixels", parent_size);
 
   pos(0)--;
   pos(1)--;
   pos(1) = parent_size(1) - pos(1) - pos(3);
 
